packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmImpl.kt

```diff
@@ -138,6 +137,20 @@ public class RealmImpl private constructor(
             }
 
             realmScope.launch {
+                configuration.encryptionKeyAsCallback?.let {
+                    // if we're using an encryption key as a callback, we preemptively open the notifier and writer Realm
+                    // with the given configuration because the key might be deleted from memory after the Realm is open.
+
+                    // These touches the notifier and writer lazy initialised Realms to open them with the provided configuration.
+                    launch(notificationScheduler.dispatcher) {
+                        notifier.realm.version().version
+                    }
+                    launch(writeScheduler.dispatcher) {
+                        writer.realm.version().version
+                        it.releaseKey()
```

Comment:
maybe wait for modifier job and the writer job to complete before calling `it.releaseKey()`